api/server/v1/marshaler.go

```diff
@@ -169,6 +173,8 @@ func (x *SearchRequest) UnmarshalJSON(data []byte) error {
 			v = &x.Project
 		case "collection":
 			v = &x.Collection
+		case "branch":
```

Comment:
SearchRequest happens on a search index, not on a branch?

Reply:
This is for the existing search API on collections. This whole SearchRequest will eventually be deprecated or evolved. But for now, we need it.